walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -82,6 +102,30 @@ private static TimeRange rangeToTimeRange(Range<Integer> range) {
     return timeRange;
   }
 
+  // helper function converting an ArrayList of TimeRange to a RangeSet of Integer
+  private static RangeSet<Integer> timeRangeToRangeSet(List<TimeRange> meetingTimes) {
+    RangeSet<Integer> timeRangeSet = TreeRangeSet.create();
+    for (TimeRange time : meetingTimes) {
+      if (time.end() != TimeRange.END_OF_DAY) {
+        timeRangeSet.add(Range.closedOpen(time.start(), time.end()));
+      } else {
+        timeRangeSet.add(Range.closed(time.start(), TimeRange.END_OF_DAY));
+      }   
+    }
+    return timeRangeSet;
+  }
+
+  // helper function that makes a deep copy of an ArrayList of TimeRanges
+  private static List<TimeRange> copyListTimeRnage(List<TimeRange> meetingTimes) {
+    // TODO: Use stream to make a deep copy
+    List<TimeRange> optionalMeetingTimes = new ArrayList<TimeRange>();
+    for (int i = 0; i < meetingTimes.size(); i++) {
+      optionalMeetingTimes.add(meetingTimes.get(i).fromStartDuration(meetingTimes.get(i).start(), 
+          meetingTimes.get(i).duration()));
```

Comment:
FYI, in production code, it might be preferable to add a

public static copyOf(TimeRange timeRange)

factory method to the TimeRange class. A common strategy for doing this with classes you don't own (e.g. if TimeRange was some other team's class and it doesn't make sense for you to modify it, or it's some third-party library code and you don't want to fork from their standard releases), is to add a utility class for adding the functionality.

e.g. `TimeRanges` which packages a bunch of helpers. In this case, a lot of helpers could fall into this class.

No action needed here, but just commenting to give you context for larger projects :)